be/src/storage/lake/vacuum.cpp

```diff
@@ -347,12 +347,13 @@ static Status collect_files_to_vacuum(TabletManager* tablet_mgr, std::string_vie
     }
     auto t1 = butil::gettimeofday_ms();
     g_metadata_travel_latency << (t1 - t0);
-
-    *vacuumed_version = final_retain_version;
     if (!skip_check_grace_timestamp) {
         // All tablet metadata files encountered were created after the grace timestamp, there were no files to delete
+        // The vacuumed_version will be set to min garbage version
+        *vacuumed_version = version;
         return Status::OK();
     }
+    *vacuumed_version = final_retain_version;
```

Comment:
shall update this version in the coming for loop in case the deletion errors in the middle?

Reply:
but this vacuumed_version is updated prior the deletion at line 359.

Reply:
> but this vacuumed_version is updated prior the deletion at line 359.

Just as stated in the previous comment submission, this vacuum version is just a temporary variable. If an error occurs during the deletion process, the vacuum version returned to the FE will not be updated(line 398-400).

Reply:
or should this version be `*vacuumed_version = final_retain_version - 1`, because the `final_retain_version` is not deleted under any case, may not be proper to take it as the `vacuumed_version`.

Reply:
> or should this version be `*vacuumed_version = final_retain_version - 1`, because the `final_retain_version` is not deleted under any case, may not be proper to take it as the `vacuumed_version`.

Agree with you. PTAL again.